specs/swagger.yml

```diff
@@ -4802,3 +5051,80 @@ definitions:
 
   specBodyDetail:
     type: string
+
+  listRoleAttributesEnvelope:
+    type: object
+    properties:
+      meta:
+        $ref: '#/definitions/meta'
+      data:
+        $ref: '#/definitions/roleAttributesList'
+
+  roleAttributesList:
+    description: An array of role attributes
+    type: array
+    items:
+      type: string
+
+  getIdentityPolicyAdviceEnvelope:
+    type: object
+    properties:
+      meta:
+        $ref: '#/definitions/meta'
+      data:
+        $ref: '#/definitions/policyAdvice'
+
+  policyAdvice:
+    type: "object"
+    required:
+      - "identityId"
+      - "identity"
+      - "serviceId"
+      - "service"
+      - "isBindAllowed"
+      - "isDialAllowed"
+      - "identityRouterCount"
+      - "serviceRouterCount"
+      - "commonRouters"
+    properties:
+      identityId:
+        type: "string"
+      identity:
+        $ref: '#/definitions/entityRef'
+      serviceId:
+        type: "string"
+      service:
+        $ref: '#/definitions/entityRef'
+      isBindAllowed:
+        type: "boolean"
+      isDialAllowed:
+        type: "boolean"
+      identityRouterCount:
+        type: "number"
+      serviceRouterCount:
+        type: "number"
+      commonRouters:
+        type: "array"
+        items:
+          type: "object"
+          properties:
+            entity:
+              type: "string"
+            id:
+              type: "string"
+            name:
+              type: "string"
+            _links:
```

Comment:
`_links` should be 

```      _links:
        $ref: '#/definitions/links'```

Reply:
They are entityRefs extended by the IsOnline field, so didn't think I could use entityRef

Reply:
oof. I didn't see isOnline there ate the bottom.

Is that necessary? The way this will work w/ generators will be an anonymous struct/class will be generated. Depending on the language this may or may not be great.

Reply:
I need the IsOnline associated with the router. I could structure it differently, so that we've got

```
commonRouters : [
    {
        isOnline: true,
        router: {
            // entityRef here
        }
    }
]
```
Which feels weird. Could I just extract out that definition as CommonRouterEntityRef or something like that so that it's named?

Reply:
You can define a generic map of additional attributes.

We would update the entityRef definition to had the following property:

```
  entityRef:
    //...existing def
    properties:
      //exisiting def
      additionalPropertiesThatYouMayCareAbout:
        type: object
        properties:
          isOnline:
            type: boolean
          someOtherProperty:
            type: string
               
```

Reply:
Thinking about this more, we can just add a generic data map to entityRef:



```
entityRef:
  //existing
  properties:
    //existing
    data:
      type: object
      additionalProperties:
        $ref: '#/definitions/dataOptions'
```